vim25/types/copy.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright (c) 2024-2024 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import "bytes"
+
+// Copy creates a copy of src into dst by encoding src using the vim
+// JSON encoder to encode src into JSON, then decoding the JSON into
+// dst.
+func Copy[T AnyType](src T) (T, error) {
```

Comment:
Ah, we do this in simulator too, but using xml: https://github.com/vmware/govmomi/blob/0e7dff84da2ce6e6ef77c5c1bedfb827a4c956ae/simulator/object.go#L67-L68